### DISCUSSION TITLE
Replace electron with tauri?

### DISCUSSION THREAD

**Discussion by ghost** [FreeTubeApp/FreeTube#2392] [General]
hi, it occurred to me that freetube uses electron frontend and I was wondering if @PrestonN was planning in the future to replace electron with tauri as it is less resource intensive, written in rust and does not load all chromium at startup because it uses the native web-engine of the system on which it is run, at the moment tauri supports distribution on the following platforms:
- Windows
- macOS
- Linux

future support:
- iOS/iPadOS
- Android

I don't list all the strengths and weaknesses I just tell you that the official site is tauri.app and the source code can be found on [Github 
tauri-apps](https://github.com/tauri-apps)

**Comment by GilgusMaximus** [comment]
Okay for the moment we are not considering that, as it will probably take a lot of effort. Additionally I believe tauri is still in hasty development, so some stuff might not work properly yet. If the work effort of porting will not turn out too big, there might be the possibility of changing to tauri, but for the moment in general tauri is very new and most team members, including me have very limited understanding of it and it's architecture.

**Comment by ghost** [comment]
thank you very much GilgusMaximus for the answer, it was just my curiosity. I fully understand the possible problems that could occur by relying on a project with a first stable version just released. I had this question because I have recently come to know and I wanted to ask some experts if it was worth proposing this new alternative that promises performance benefits and reduced dimensions of the compiled output. Thank you again for your patience.

> And that basically is the problem with tauri. We won't rewrite the whole application backend logic + all the dependencies we have on backend packages in rust. 

Ps. however you shouldn't rewrite anything in rust as the code should remain the same as freetube.

**Comment by 4NUB1S-0N-0P10M** [comment]
I have a very good idea regarding this concern. How about creating a new port for freetube to work with Tauri as beta, then fix any problems that comes with it overtime? 
The problem is that If there won't be an initiative to evolve with new frameworks (regardless if they aren't stable enough for development), that wouldn't make it possible to improve freetube with better features and more benefits. Apart from the bloatware issue that Electron faces, many users dislike using Javascript-based programs (including me), and i believe that Tauri fixes most of these issues if done properly and efficiently. Hopefully that was enough to emphasize the demand/reasoning for considering the different alternatives out there.
